std/src/main/scala/algebra/std/map.scala

```diff
@@ -30,23 +32,98 @@ class MapAdditiveMonoid[K, V](implicit V: AdditiveSemigroup[V]) extends Additive
 
   def plus(xs: Map[K, V], ys: Map[K, V]): Map[K, V] =
     if (xs.size <= ys.size) {
-      addMap(xs, ys)((x, y) => V.plus(x, y))
+      xs.foldLeft(ys) { case (my, (k, x)) =>
+        my.updated(k, my.get(k) match {
+          case Some(y) => V.plus(x, y)
+          case None => x
+        })
+      }
     } else {
-      addMap(ys, xs)((y, x) => V.plus(x, y))
+      ys.foldLeft(xs) { case (mx, (k, y)) =>
+        mx.updated(k, mx.get(k) match {
+          case Some(x) => V.plus(x, y)
+          case None => y
+        })
+      }
+    }
+
+  override def sumN(a: Map[K, V], n: Int): Map[K, V] =
+    if (n > 0) a.map { case (k, v) => (k, V.sumN(v, n)) }
+    else if (n == 0) zero
+    else throw new IllegalArgumentException("Illegal negative exponent to sumN: %s" format n)
+
+  override def sum(as: TraversableOnce[Map[K, V]]): Map[K, V] = {
```

Comment:
we don't have `trySum` here:

https://github.com/non/algebra/blob/master/ring/src/main/scala/algebra/ring/Additive.scala#L32

can we do the same override on AdditiveMonoid or here so that `trySum` calls `sum`?